### PR TITLE
Merge and solve conflicts `expire at` and `options` with fix bugs

### PR DESCRIPTION
**Pull request by alireaza**

### CODE REVIEW

**Review by mohsen-farahani — APPROVED** [review]

**Review by mohsen-farahani — APPROVED** [review]

**Review by mohsen-farahani — DISMISSED** [review]

File: src/NotificationProviders/SmsProviders/Smsir/SmsirProvider.php [hunk 1]
```diff
@@ -10,79 +10,87 @@ class SmsirProvider extends SmsAbstract
...
-            $result["result_id"] = null;
-            $result["errors"] = $response["Message"];
+            $result["result_id"] = NULL;
+            $result["errors"]    = $response["Message"];
         }
 
         return $result;
     }
+
+    public function getToken()
```
Comment (Rasoul-Karimi) [inline src/NotificationProviders/SmsProviders/Smsir/SmsirProvider.php:73]: add return type 
notice: in catch bloc you  return boolean and in method body return array .. so notice in return type

File: src/Notifier.php [hunk 2]
```diff
@@ -11,27 +11,43 @@
 
 class Notifier
 {
-    public static function sendPush(string $heading, string $content, array $player_ids, array $extra = null)
+    private static $expire_at = 0;
+
+    private static $options = [];
+
+    public static function sendPush(string $heading, string $content, array $player_ids, array $extra = NULL)
```
Comment (Rasoul-Karimi) [inline src/Notifier.php:18]: add return type

File: src/Notifier.php [hunk 3]
```diff
@@ -11,27 +11,43 @@
...
     {
-        dispatch(new SendPushJob($heading, $content, $player_ids, $extra));
+        dispatch(new SendPushJob($heading, $content, $player_ids, $extra, static::$expire_at, static::$options));
 
-        return true;
+        return TRUE;
     }
 
     public static function sendSms(string $message, array $numbers)
```
Comment (Rasoul-Karimi) [inline src/Notifier.php:25]: same as

File: src/Notifier.php [hunk 4]
```diff
@@ -11,27 +11,43 @@
...
 {
-    public static function sendPush(string $heading, string $content, array $player_ids, array $extra = null)
+    private static $expire_at = 0;
+
+    private static $options = [];
+
+    public static function sendPush(string $heading, string $content, array $player_ids, array $extra = NULL)
     {
-        dispatch(new SendPushJob($heading, $content, $player_ids, $extra));
+        dispatch(new SendPushJob($heading, $content, $player_ids, $extra, static::$expire_at, static::$options));
```
Comment (Rasoul-Karimi) [inline src/Notifier.php]: in static class in oop solution, if you need to select or use static variable you need to use self::VariableName
and if you need to create static object of class need to Write static::

in final: static:: it's for create method and variable but self:: its for call them

File: src/Notifier.php [hunk 5]
```diff
@@ -11,27 +11,43 @@
...
 
-        return true;
+        return TRUE;
     }
 
     public static function sendSms(string $message, array $numbers)
     {
-        dispatch(new SendSmsJob($message, $numbers));
+        dispatch(new SendSmsJob($message, $numbers, static::$expire_at, static::$options));
```
Comment (Rasoul-Karimi) [inline src/Notifier.php]: use self for use static method or variable

File: src/Traits/MessageTrait.php [hunk 6]
```diff
@@ -3,40 +3,50 @@
...
 {
-    protected $current_provider = null;
+    private $current_provider = NULL;
+    private $title;
+    private $body;
+    private $user_ids;
+    private $expire_at;
+    private $options;
 
     public function sendMessage()
```
Comment (Rasoul-Karimi) [inline src/Traits/MessageTrait.php:18]: add return type

File: src/Traits/SmsTrait.php [hunk 7]
```diff
@@ -8,34 +8,45 @@
...
 {
-    protected $current_provider = null;
+    private $current_provider = NULL;
+    private $message;
+    private $numbers;
+    private $datetime;
+    private $expire_at;
+    private $options;
 
     public function sendSms()
```
Comment (Rasoul-Karimi) [inline src/Traits/SmsTrait.php:18]: add return type

File: src/Traits/PushTrait.php [hunk 8]
```diff
@@ -8,50 +8,58 @@
...
-    protected $current_provider = null;
+    private $current_provider = NULL;
+    private $heading;
+    private $content;
+    private $player_ids;
+    private $extra;
+    private $expire_at;
+    private $options;
 
     public function sendPush()
```
Comment (Rasoul-Karimi) [inline src/Traits/PushTrait.php:19]: add return type

File: src/Notifier.php [hunk 9]
```diff
@@ -11,27 +11,43 @@
...
+    }
+
+    public static function setExpireAt($expire_at)
+    {
+        static::$expire_at = $expire_at;
+    }
+
+    public static function options(array $options = [])
+    {
+        static::$options = $options;
```
Comment (Rasoul-Karimi) [inline src/Notifier.php]: use self same as..

File: src/Notifier.php [hunk 10]
```diff
@@ -11,27 +11,43 @@
 
 class Notifier
 {
-    public static function sendPush(string $heading, string $content, array $player_ids, array $extra = null)
+    private static $expire_at = 0;
```
Comment (Rasoul-Karimi) [inline src/Notifier.php]: use self::

File: src/Notifier.php [hunk 11]
```diff
@@ -11,27 +11,43 @@
...
+
+    private static $options = [];
```
Comment (Rasoul-Karimi) [inline src/Notifier.php]: use self::

**Review by Mehrdad-Dadkhah — DISMISSED** [review]

**Review by Mehrdad-Dadkhah — APPROVED** [review]

**Pull request merged**